File: 7_Reverse_Integer/reverse.py
class Solution(object):
    def reverse(self, x):
        if not x: return x
        digits = []
        sign = x / abs(x)
        x = abs(x)
        while x:
            digits.append(x % 10)
            x /= 10

        res = 0
        length = len(digits)
        for idx in xrange(length):
            res += 10 ** (length - idx - 1) * digits[idx]

        return sign * res if res <= 2147483647 and res >= -2147483648 else 0

sol = Solution()
x = 1534236469
print sol.reverse(x)

File: 7_Reverse_Integer/reverse.cpp
#include <iostream>

using namespace std;

class Solution {
	public:
    	int reverse(int x) {
            /* Since res may be bigger than INT_MAX or smaller than INT_MIN. */
            long res = 0;
            int sign = x > 0? 1 : -1;
            x = abs(x);
            while(x) {
                res = res * 10 + x % 10;
                x /= 10;
            }

            return (res <= INT_MAX && res >= INT_MIN)? res * sign : 0;

    	}
};

int main() {
    return 0;
}
